src/modules/petrology/FeatureDetail.js

```diff
@@ -0,0 +1,269 @@
+    return true;
+  };
+
+  const validateAndSetNewType = (newType) => {
+    let existingCustomFeatureTypes = customFeatureTypes.filter((feature) => feature.name === newType);
+    if (!isEmpty(existingCustomFeatureTypes)) {
+      Alert.alert('Alert!',
+        'The type ' + newType + ' is already being used. Choose a different type name.');
+      setOtherType('');
+      return false;
+    }
+    else if (isEmpty(otherType)) {
+      Alert.alert('Alert!',
+        'The new type being defined is empty');
+      return false;
+    }
+    else return true;
+  };
+
+  const renderFeatureForm = () => {
+
+    // Validate the feature
+    const validateFeature = (values) => {
+      let errors = {};
+      if (values.name === '') errors.name = 'Feature name cannot be empty';
+      return errors;
+    };
+
+    const initialFeatureValues = {
+      label: selectedFeature.label ? selectedFeature.label : '',
+      name: selectedFeature.name ? selectedFeature.name : '',
+      type: selectedFeature.type ? selectedFeature.type : '',
+      description: selectedFeature.description ? selectedFeature.description : '',
+    };
+
+    return (
+      <View style={{flex: 1}}>
+        <Formik
+          initialValues={initialFeatureValues}
+          onSubmit={onSubmitForm}
+          validate={validateFeature}
+          innerRef={featureForm}
+          validateOnChange={true}
+          enableReinitialize={true}
+        >
+          {() => (
+            <View>
+              <ListItem containerStyle={commonStyles.listItemMinimalVerticalPadding}>
+                <ListItem.Content>
+                  <Field
+                    component={TextInputField}
+                    name={'label'}
+                    label={'Label'}
+                    key={'label'}
+                  />
+                </ListItem.Content>
+              </ListItem>
+              <FlatListItemSeparator/>
+              <ListItem containerStyle={commonStyles.listItemMinimalVerticalPadding}>
+                <ListItem.Content>
+                  <Field
+                    component={TextInputField}
+                    name={'name'}
+                    label={'Name'}
+                    key={'name'}
+                  />
+                </ListItem.Content>
+              </ListItem>
+              <FlatListItemSeparator/>
+              <ListItem containerStyle={commonStyles.listItemMinimalVerticalPadding}>
+                <ListItem.Content>
+                  <Field
+                    component={TextInputField}
+                    name={'description'}
+                    label={'Feature Description'}
+                    key={'description'}
+                  />
+                </ListItem.Content>
+              </ListItem>
+              <ListItem containerStyle={commonStyles.listItemMinimalVerticalPadding}>
+                <ListItem.Title style={formStyles.fieldLabel}>{'Feature Type'}</ListItem.Title>
+              </ListItem>
+              <MultiSelect
```

Comment:
Can you use the component SelectInputField here instead? See https://github.com/StraboSpot/StraboSpot-Mobile/blob/3ec1ec57b32cea70e898fdbe521c20186e2443b8/src/modules/petrology/RockSubpage.js#L113